llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -2876,42 +2875,111 @@ void AsmPrinter::emitJumpTableInfo() {
       MJTI->getEntryKind() == MachineJumpTableInfo::EK_LabelDifference32 ||
           MJTI->getEntryKind() == MachineJumpTableInfo::EK_LabelDifference64,
       F);
+
+  std::vector<unsigned> JumpTableIndices;
```

Comment:
I think the attempt to reuse this vector for both cases - with and without partitioning makes the code more complex than it needs to be. Consider the following refactoring:
* Sink the vector into the if condition below.
* Use separate vectors for hot and cold when partitioning is enabled.
* Use an arrayref as param for emitJumpTableImpl instead of iterator range.